### PR TITLE
Add EIP: ETH transfers emit a log

### PR DESCRIPTION
**Pull request by vbuterin**
All ETH-transferring calls emit a log.

### CODE REVIEW

File: EIPS/eip-9999.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+eip: 9999
```
Comment (abcoathup) [inline EIPS/eip-9999.md:2]:
```suggestion
eip: 7708
```
Assigning next sequential EIP/ERC/RIP number.

Please also update the filename.

File: EIPS/eip-9999.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+title: ETH transfers emit a log
+description: All ETH transfers emit a log
+author: Vitalik Buterin (@vbuterin), Peter Davies (@petertdavies)
+discussions-to: https://ethereum-magicians.org/t/eip-all-value-transferring-calls-create-a-log/20034
```
Comment (abcoathup) [inline EIPS/eip-9999.md:6]:
```suggestion
discussions-to: https://ethereum-magicians.org/t/eip-7708-eth-transfers-emit-a-log/20034
```

File: EIPS/eip-9999.md (hunk 1 above)
Comment (finewhitechic) [reply EIPS/eip-9999.md:2]: Oh ok i don't have a clue on what to do here. I'm just trying to figure out how to borrow usd. With my collateral. Tokens.

**Review by finewhitechic — APPROVED** [review]

**Review by Hawthorne001 — APPROVED** [review]
tests

**Comment by finewhitechic** [comment]
> All ETH-transferring calls emit a log.

I want to make a transfer with my tokens. To borrow usd.  I'm not sure what to do or how to go by doing it???

File: EIPS/eip-7708.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+---
+eip: 7708
+title: ETH transfers emit a log
+description: All ETH transfers emit a log
+author: Vitalik Buterin (@vbuterin), Peter Davies (@petertdavies)
+discussions-to: https://ethereum-magicians.org/t/eip-all-value-transferring-calls-create-a-log/20034
```
Comment (vbuterin) [inline EIPS/eip-7708.md:6]:
```suggestion
discussions-to: https://ethereum-magicians.org/t/eip-7708-eth-transfers-emit-a-log/20034
```

**Comment by holic** [comment]
my kingdom for a log on CREATE as well

**Comment by cstoneham** [comment]
There's nothing I desire more than this.

**Comment by karalabe** [comment]
Would emitting thee new logs be free or incur the default log emit costs? Asking because if it costs something, then gas accounting can get a bit unky in certain places (e.g. if I want to emit a log with the total fees paid, which depend on emitting the log itself, before emitting the log itself :P)

**Comment by charlieflipside** [comment]
> All ETH-transferring calls emit a log.

Would emphasize non-traditional transfers need (possibly duplicative) logs as well to ensure the UX assumptions of transfers holds. Adding ETH transfer logs without these special cases handled will result in ethereum data providers to be inundated with faulty assumptions from their users (most specifically, net transfers != ETH balance at a block height)

ETH Gas payments should also emit event that corresponds with a transfer to 0x0000...0000
emit gas_paid(user, amt)
     emit transfer(from user, to 0x0000...0000)
     
ETH beacon chain rewards should emit 
emit beacon_mint (receiver, amt)
    emit transfer(from 0x000...0000, to user)

contract self destructs should emit 2 transfers 
    emit self_destruct (contract, to, amt)
    emit transfer(from contract, to 0x000...0000)
    emit transfer(from 0x000....0000, to receiver)
    
    
   Data providers currently use traces to identify ETH transfers, so logs would be a small quality of life improvement if done well, but could actually cause more headaches if special cases (like self destruct bypassing contract `payable()` status) are not handled.

**Comment by fergmolina** [comment]
agree with @charlieflipside . it should include fees payments and block rewards. My only concern here is that this change is going to almost duplicate the amount of logs per day: [Dune Dashboard - EIP-7708](https://dune.com/fergmolina/eip-7708-eth-transfers-emit-a-log?start_date_d373f8=2024-01-01+00%3A00%3A00&end_date_dff03c=)

**Comment by Selarun15** [comment]
0xa170ecfc733473C7c757185fEb279Bf282D83CDa

**Comment by Selarun15** [comment]
0xa170ecfc733473C7c757185fEb279Bf282D83CDa

**Comment by PHOENIXENT** [comment]
[wwdc2023-10042 2.txt](https://github.com/user-attachments/files/15601750/wwdc2023-10042.2.txt)

**Review by Hawthorne001 — APPROVED** [review]
Signed off by: tylerhawthorne@MainH.Onmicrosoft.com

File: EIPS/eip-9999.md (hunk 2 above)
Comment (Selarun15) [reply EIPS/eip-9999.md]: > 

Okey

**Review by Selarun15 — APPROVED** [review]

File: EIPS/eip-9999.md (hunk 2 above)
Comment (Selarun15) [reply EIPS/eip-9999.md]: 0xa170ecfc733473C7c757185fEb279Bf282D83CDa

**Review by g11tech — APPROVED** [review]
lgtm for draft, some Todos and open questions that would need to be addressed before review

**Review by nichechristie — COMMENTED** [review]
Nichebiche

**Review by nichechristie — COMMENTED** [review]
Nichebiche

**Comment by KillariDev** [comment]
Ethereum transfers generating logs is super useful! In `eth_simulateV1` (https://github.com/ethereum/execution-apis/pull/484) we decided to add a flag `traceTransfers` to enable ETH logs. This flag will trace Ethereum transfers and return them as ERC20 transfer logs:
```
"logs": [
	{
		"address": "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
		"topics": [
			"0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
			"0x000000000000000000000000c000000000000000000000000000000000000000",
			"0x000000000000000000000000c100000000000000000000000000000000000000"
		],
		"data": "0x00000000000000000000000000000000000000000000000000000000000003e8",
		"blockNumber": "0x4",
		"transactionHash": "0xa4d41019e71335f8567e17746b708ddda8b975a9a61f909bd3df55f4866cc913",
		"transactionIndex": "0x0",
		"blockHash": "0x859c932c5cf0dabf8d12eb2518e063966ac1a25e2fc49f1f02574a37f358d0b5",
		"logIndex": "0x0",
		"removed": false
	}
],
```
you can see the log coming from address `0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee`, these are eth transfer logs.

These logs are emitted when you simulate a transaction with `eth_simulateV1`. These logs don't contribute to the gas cost, and they are not included in logs bloom in `eth_simulateV1`. This is of course only a simulation (like `eth_call`, these logs are not stored anywhere after the call has finished.

**Review by nichechristie — COMMENTED** [review]
Awesome niche

**Comment by PHOENIXENT** [comment]
`9x9876543210.0123456789!`

**Comment by Nichebiche** [comment]
> 9x9876543210.0123456789!

**Comment by HoangTranMap** [comment]
??

Vào Th 6, 28 thg 6, 2024 vào lúc 22:40 Nichebiche <
***@***.***> đã viết:

**Comment by phadungkiat627** [comment]
> ✅ All reviewers have approved.

**Review by nigelfpowers — COMMENTED** [review]
oops

**Review by 7zkm7b8gw9-web — COMMENTED** [review]
Whats next?

**Comment by Musthaq22** [comment]
vbuterin-patch-2

**Review by Musthaq22 — COMMENTED** [review]
vbuterin-patch-2

**Review by Musthaq22 — COMMENTED** [review]
6c02ac6b7bd28c91656ce870390624d962589ff5

**Review by Musthaq22 — COMMENTED** [review]
136f03559e0e710f284b41d15c855e1be7c77f67

**Comment by sin-strokes** [comment]
Ether transfer

**Pull request merged**